src/Microsoft.TestPlatform.Common/ExtensionFramework/VSExtensionManager.cs

```diff
@@ -0,0 +1,203 @@
+
+            // Check for devenv
+            if (!this.fileHelper.Exists(this.pathToDevenv))
+            {
+                throw new TestPlatformException(string.Format(CultureInfo.CurrentCulture, Resources.VSInstallationNotFound, this.vsInstallPath));
+            }
+
+            // Adding resolution paths for resolving dependencies.
+            var resolutionPaths = new List<string>() {
+                this.vsInstallPath,
+                Path.Combine(this.vsInstallPath, PrivateAssembliesDirName),
+                Path.Combine(this.vsInstallPath, PublicAssembliesDirName)
+            };
+            var assemblyResolver = new AssemblyResolver(resolutionPaths);
+
+            object extensionManager;
+            object settingsManager;
+                                       
+            settingsManager = SettingsManagerType.GetMethod("CreateForApplication", new Type[] { typeof(String) }).Invoke(null, new object[] { pathToDevenv });
+            if (settingsManager != null)
+            {
+                // create extension manager
+                extensionManager = Activator.CreateInstance(ExtensionManagerServiceType, settingsManager);
+
+                if (extensionManager != null)
+                {
+                    installedExtensions = ExtensionManagerServiceType.GetMethod("GetEnabledExtensionContentLocations", new Type[] { typeof(String) }).Invoke(
+                                               extensionManager, new object[] { extensionType }) as IEnumerable<string>;
+                }
+            
+                // Dispose the settings manager
+                IDisposable disposable = (settingsManager as IDisposable);
+                if (disposable != null)
+                {
+                    disposable.Dispose();
+                }
+            }
+
+            // Unregister the resolver by disposing
+            assemblyResolver.Dispose();
+
+            return installedExtensions;
+        }
+
+        /// <summary>
+        /// Used to explicitly load Microsoft.VisualStudio.ExtensionManager.dll
+        /// </summary>
+        private Assembly ExtensionManagerDefAssembly
+        {
+            get
+            {
+                if (extensionManagerAssembly == null)
+                {
+                    extensionManagerAssembly = Assembly.Load(new AssemblyName(ExtensionManagerAssemblyName));
+                }
+                return extensionManagerAssembly;
+            }
+        }
+
+        /// <summary>
+        /// Used to explicitly load Microsoft.VisualStudio.ExtensionManager.Implementation.dll
+        /// </summary>
+        private Assembly ExtensionManagerImplAssembly
+        {
+            get
+            {
+                if (extensionManagerImplAssembly == null)
+                { 
+                    // Make sure ExtensionManager assembly is already loaded.
+                    Assembly a = ExtensionManagerDefAssembly;
```

Comment:
If `a` is not used, is it possible that compiler optimizes it away?

Reply:
Makes sense, will turn this into check for null.